src/github.com/matrix-org/go-neb/services/github/github.go

```diff
@@ -219,6 +221,12 @@ func (s *githubService) Register() error {
 }
 
 func (s *githubService) PostRegister(oldService types.Service) {
+	// PostRegister handles creating/destroying webhooks, which is only valid if this service
+	// is configured on behalf of a client.
+	if s.ClientUserID == "" {
+		return
```

Comment:
That makes no sense. `PostRegister` is called regardless of whether any webhooks are set.